src/compiler/checker/lines_32000_40000.rs

```diff
@@ -318,6 +319,20 @@ impl TypeChecker {
         self.get_type_from_type_node(node);
     }
 
+    pub(super) fn check_block(&mut self, node: &Block) {
+        if is_function_or_module_block(node) {
+            for_each(&node.statements, |statement, _| {
+                self.check_source_element(Some(statement.clone()));
+                Option::<()>::None
```

Comment:
What's this for?

Reply:
The `Option` return type of the callback is generic so it complains if it can't resolve it to a specific type